src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -942,25 +944,231 @@ private void DecodeModuleInitializerAttribute(DecodeWellKnownAttributeArguments<
-                throw ExceptionUtilities.Unreachable();
+                DecodeInterceptsLocationAttributeExperimentalCompat(arguments, attributeFilePath: (string?)constructorArguments[0].Value, lineNumberOneBased, characterNumberOneBased);
+            }
+            else
+            {
+                Debug.Assert(arguments.Attribute.AttributeConstructor.Parameters is [{ Type.SpecialType: SpecialType.System_Int32 }, { Type.SpecialType: SpecialType.System_String }]);
+                DecodeInterceptsLocationChecksumBased(arguments, version: (int)constructorArguments[0].Value!, data: (string?)constructorArguments[1].Value);
+            }
+        }
+
+        private void DecodeInterceptsLocationChecksumBased(DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments, int version, string? data)
+        {
+            var diagnostics = (BindingDiagnosticBag)arguments.Diagnostics;
+            if (version != 1)
+            {
+                diagnostics.Add(ErrorCode.ERR_InterceptsLocationUnsupportedVersion, arguments.Attribute.GetAttributeArgumentLocation(parameterIndex: 0));
+                return;
+            }
+
+            if (data is null)
+            {
+                diagnostics.Add(ErrorCode.ERR_InterceptsLocationDataInvalidFormat, arguments.Attribute.GetAttributeArgumentLocation(parameterIndex: 1));
+                return;
+            }
+
+            byte[] bytes;
+            try
+            {
+                bytes = Convert.FromBase64String(data);
+            }
+            catch (FormatException)
+            {
+                diagnostics.Add(ErrorCode.ERR_InterceptsLocationDataInvalidFormat, arguments.Attribute.GetAttributeArgumentLocation(parameterIndex: 1));
+                return;
+            }
+
+            // format:
+            // - 16 bytes of target file content hash (xxHash128)
+            // - int32 position (little endian)
+            // - utf-8 display filename
+            const int hashIndex = 0;
+            const int hashSize = 16;
+            const int positionIndex = hashIndex + hashSize;
+            const int positionSize = sizeof(int);
+            const int displayNameIndex = positionIndex + positionSize;
+            const int minLength = displayNameIndex;
+
+            if (bytes.Length < minLength)
+            {
+                diagnostics.Add(ErrorCode.ERR_InterceptsLocationDataInvalidFormat, arguments.Attribute.GetAttributeArgumentLocation(parameterIndex: 1));
+                return;
+            }
+
+            var hash = bytes.AsMemory(start: hashIndex, length: hashSize);
+            var position = BinaryPrimitives.ReadInt32LittleEndian(bytes.AsSpan(start: positionIndex));
+
+            string displayFileName;
+            try
+            {
+                displayFileName = Encoding.UTF8.GetString(bytes, index: displayNameIndex, count: bytes.Length - displayNameIndex);
+            }
+            catch (ArgumentException)
```

Comment:
Do we have a test covering this case (fails to decode as UTF8)?

Reply:
It's odd, I did introduce `Checksum_03` for this purpose, and at some point I was sure it was actually throwing an exception here. But now it seems to be using � instead to denote the invalid characters, both in framework and core.

Since GetString is documented to be able to throw this here, I'll keep the try/catch, and am open to any further suggestions on how to get it to actually throw in a test.

Reply:
According to ChatGPT, some byte sequences that should fail: `{ 0xC3 }`, `{ 0xC0, 0xAF }`, `{ 0xC2, 0x20 }`, `{ 0xFF }`. Also, we may need to check what is the "error detection" behavior (`Fallback` property) of the decoder (a decoder with a fallback could replace bad characters).

Reply:
Looked further at this and am adjusting the impl here based on the documented guarantees of `Encoding.UTF8`, which I'll include in comments here.